helix-rest/src/test/java/org/apache/helix/rest/metadatastore/TestTrieRoutingData.java

```diff
@@ -19,146 +19,180 @@
  * under the License.
  */
 
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.NoSuchElementException;
+import org.apache.helix.rest.metadatastore.exceptions.InvalidRoutingDataException;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
 public class TestTrieRoutingData {
-  // TODO: add constructor related tests after constructor is finished
+  private TrieRoutingData _trie;
 
   @Test
-  public void testGetAllMappingUnderPathFromRoot() {
-    TrieRoutingData trie = constructTestTrie();
-    Map<String, String> result = trie.getAllMappingUnderPath("/");
-    Assert.assertEquals(result.size(), 4);
-    Assert.assertEquals(result.get("/b/c/d"), "realmAddressD");
-    Assert.assertEquals(result.get("/b/c/e"), "realmAddressE");
-    Assert.assertEquals(result.get("/b/f"), "realmAddressF");
-    Assert.assertEquals(result.get("/g"), "realmAddressG");
+  public void testConstructionSpecialCase() {
```

Comment:
Do you think you could add more descriptions about each test case?

Reply:
Ack